x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/SqlMediaTypeParser.java

```diff
@@ -34,27 +36,47 @@
      * isn't but there is a {@code Accept} header then we use that. If there
      * isn't then we use the {@code Content-Type} header which is required.
      */
-    public MediaType getResponseMediaType(RestRequest request, SqlQueryRequest sqlRequest) {
+    public static MediaType getResponseMediaType(RestRequest request, SqlQueryRequest sqlRequest) {
         if (Mode.isDedicatedClient(sqlRequest.requestInfo().mode())
             && (sqlRequest.binaryCommunication() == null || sqlRequest.binaryCommunication())) {
             // enforce CBOR response for drivers and CLI (unless instructed differently through the config param)
             return XContentType.CBOR;
         } else if (request.hasParam(URL_PARAM_FORMAT)) {
             return validateColumnarRequest(sqlRequest.columnar(),
-                MEDIA_TYPE_REGISTRY.queryParamToMediaType(request.param(URL_PARAM_FORMAT)));
+                MEDIA_TYPE_REGISTRY.queryParamToMediaType(request.param(URL_PARAM_FORMAT)), request);
         }
 
-        if (request.getParsedAccept() != null) {
-            return request.getParsedAccept().toMediaType(MEDIA_TYPE_REGISTRY);
+        return getResponseMediaType(request);
+    }
+
+    public static MediaType getResponseMediaType(RestRequest request) {
```

Comment:
This method should delegate to the one above or vice-versa. 
The `URL_PARAM_FORMAT` & co is shared across the two.

Reply:
The check on `URL_PARAM_FORMAT` needs to stay in both functions, since the functions are called from different context and need to act differently on its presence. But I've refactored the functions to avoid the double check.